typescript/scripts/build.sh

```diff
@@ -13,4 +13,12 @@ lang_root_dirpath="$(dirname "${script_dirpath}")"
 # # ==================================================================================================
 # #                                             Main Logic
 # # ==================================================================================================
-find "${lang_root_dirpath}/mainLogic" -name "*.ts" -exec tsc --out "${lang_root_dirpath}/build" {} \;
+#find "${lang_root_dirpath}/lib" -name "*.ts" -exec tsc --outFile "${lang_root_dirpath}/build/output.js" --module system --moduleResolution node {} \;
+
+totalFiles=""
+while read -r line
+do
+        totalFiles="$totalFiles$line "
+done <<< "$(find "${lang_root_dirpath}/lib" -name "*.ts")"
+
+tsc $totalFiles --outFile "${lang_root_dirpath}/build/output.js" --module system --moduleResolution node
```

Comment:
* `--out` is deprecated and it was preferred that I used `--outFile` instead 
* `--module` was needed in order to avoid the following error with my `export type ServiceID = string;` => `Cannot compile modules using option 'out' unless the '--module' flag is 'amd' or 'system'.`
* `--moduleResolution` was needed after I changed module from its default `commonJS` to `system`, since it didn't recognize any of the packages in my package.json
* I added the while loop since I realized after testing in the shell and making a small bash script that my output.js file was being overwritten, and not appending each compiled .ts file into .js (I wanted to make sure that I was imitating `tsc file1.ts file2.ts ... --outFile output.js .....`)

Reply:
ah, that `commonJS` vs `system` thing is probably why your `require("path")` isn't working: https://stackoverflow.com/questions/41326485/difference-between-module-type-in-tsconfig-json/41326855

Reply:
Re. the while loop - you actually don't need this due to a tricky feature of Bash:
let's say you have a multiline string (which simulates the contents of a `find` command):
```
myvar="file1.txt
file2.txt"
```
if I do `echo "${myvar}"` then I'm going to get this output:
```
file1.txt
file2.txt
```
however, if I do `echo ${myVar}` (notice the lack of double quotes) I get:
```
file1.txt file2.txt
```
What gives??? This is a complicated (and IMO stupid, but sometimes useful) feature of shell programming known as "word-splitting".

When a variable is quoted with `""`, it will be preserved exactly as-is, so `echo "${myvar}"` is basically `echo "file1.txt\nfile2.txt"`. This is what you want 90% of the time, and by default you should always quote your variables.

However, when the `""` is omitted, Bash will chop up the string based on whitespace and _feed the separate components as different args_.
Therefore, without quoting, `echo ${myvar}` is equivalent to `echo "file1.txt" "file2.txt"`

How is that useful here? Well, `find`'s output is giving you a single string like this:
```
file1.txt
file2.txt
```
but you actuall want `file1.txt` `file2.txt` so you can feed them to `tsc`.

The solution: capture the find output, but don't quote it when you feed it to `tsc`:
```
find_results="$(find "${lang_root_dirpath}/lib" -name "*.ts")"
tsc ${find_results}
```
NOTE: This WILL break if any of the filenames have a space in it, which is why:
1) putting spaces in a Linux file is taboo
2) the `bootstrap.sh` broke when you were first onboarding, and you had `Ali Syed` as a directory name

However, it's a risk I'm comfortable taking - anyone using Linux should know better than to put spaces in filenames

Reply:
Yep I realized the require("path") wasn't working for me either, until I installed the latest version grpc via `npm install grpc` which upgraded the dependency that grpc and path were relying on inside `package-lock.json`. Then it worked just fine.